File: herocats_extras/Accepted_it.java
import java.util.Scanner;

public class Accepted_it {
    public static int performMissions(int moneyCap, int timeCap, int[][] missions) {
        int numMissions = missions.length;
        int[][] dp = new int[timeCap + 1][moneyCap + 1];

        for (int missionI = 1; missionI <= numMissions; missionI++) {
            int cost = missions[missionI - 1][0];
            int timeConsumed = missions[missionI - 1][1];
            int rescues = missions[missionI - 1][2];

            for (int t = timeCap; t >= timeConsumed; t--) {
                for (int m = moneyCap; m >= cost; m--) {
                    if (cost <= m && timeConsumed <= t) {
                        int performRescue = dp[t - timeConsumed][m - cost] + rescues;
                        int dontRescue = dp[t][m];
                        dp[t][m] = Math.max(performRescue, dontRescue);
                    }
                }
            }
        }

        return dp[timeCap][moneyCap];
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int M = scanner.nextInt();
        int T = scanner.nextInt();
        int N = scanner.nextInt();
        int[][] missions = new int[N][3];
        for (int i = 0; i < N; i++) {
            missions[i][0] = scanner.nextInt();
            missions[i][1] = scanner.nextInt();
            missions[i][2] = scanner.nextInt();
        }

        int rescues = performMissions(M, T, missions);
        System.out.println(rescues);
    }
}

File: herocats/submissions/accepted/accepted_it.c
#include <stdio.h>

int perform_missions(int money_cap, int time_cap, int num_missions, int missions[num_missions][3]) {
    int dp[time_cap + 1][money_cap + 1];

    for (int i = 0; i <= time_cap; i++) {
        for (int j = 0; j <= money_cap; j++) {
            dp[i][j] = 0;
        }
    }

    for (int mission_i = 1; mission_i <= num_missions; mission_i++) {
        int cost = missions[mission_i - 1][0];
        int time_consumed = missions[mission_i - 1][1];
        int rescues = missions[mission_i - 1][2];

        for (int t = time_cap; t >= time_consumed; t--) {
            for (int m = money_cap; m >= cost; m--) {
                if (cost <= m && time_consumed <= t) {
                    int perform_rescue = dp[t - time_consumed][m - cost] + rescues;
                    int dont_rescue = dp[t][m];
                    dp[t][m] = (perform_rescue > dont_rescue) ? perform_rescue : dont_rescue;
                }
            }
        }
    }

    return dp[time_cap][money_cap];
}

int main() {
    int M, T, N;
    scanf("%d %d %d", &M, &T, &N);

    int missions[N][3];
    for (int i = 0; i < N; i++) {
        scanf("%d %d %d", &missions[i][0], &missions[i][1], &missions[i][2]);
    }

    int rescues = perform_missions(M, T, N, missions);
    printf("%d\n", rescues);

    return 0;
}
